src/Parser/HolyPriest/Modules/PriestCore/Serendipity.js

```diff
@@ -0,0 +1,227 @@
+    [SPELLS.GREATER_HEAL.id]: SPELLS.HOLY_WORD_SERENITY.id,
+    [SPELLS.FLASH_HEAL.id]: SPELLS.HOLY_WORD_SERENITY.id,
+    [SPELLS.PRAYER_OF_HEALING.id]: SPELLS.HOLY_WORD_SANCTIFY.id,
+  }
+
+  getAffectedSpell(spellId) {
+    if (Object.keys(this.serendipityProccers).indexOf(spellId.toString()) === -1) {
+      return -1;
+    }
+    return this.serendipityProccers[spellId];
+  }
+
+  // CD tracker
+  holyWordCooldowns = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 0,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0,
+  }
+
+  // Cooldowns for Holy Words
+  holyWordMaxCooldown = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 60,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0, // modified in on_initialized
+  }
+
+  // Serendipity amount (modified accordingly in on_initialized)
+  serendipityReduction = 6.0
+
+
+  //   Output values
+
+  holy2PsetValue = 0.0
+
+  // Overall wasted serendipity
+  overcast = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 0.0,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0.0,
+  }
+  // Cast tracker
+  casts = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 0,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0,
+  }
+  // How many full casts were missed (i.e. 60s of consecutive wasted Serendipity on Serenity)
+  fullOvercast = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 0,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0,
+  }
+  // Tracker of how much wasted overcast between each holy word cast
+  _tempOvercast = {
+    [SPELLS.HOLY_WORD_SERENITY.id]: 0.0,
+    [SPELLS.HOLY_WORD_SANCTIFY.id]: 0.0,
+  }
+
+  on_initialized() {
+    // Set up proper serendipity reduction values
+    if (this.combatants.selected.hasTalent(SPELLS.LIGHT_OF_THE_NAARU_TALENT.id)) {
+      this.serendipityReduction += 2.0;
+    }
+    if (this.combatants.selected.hasBuff(SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF)) {
+      this.serendipityReduction += 1.0;
+    }
+
+    // Check for Piety for serendipityProccers
+    if (this.combatants.selected.hasTalent(SPELLS.PIETY_TALENT.id)) {
+      this.serendipityProccers[SPELLS.PRAYER_OF_MENDING_CAST.id] = SPELLS.HOLY_WORD_SANCTIFY.id;
+    }
+
+    // Modify Sanctify CD based on trait
+    this.holyWordMaxCooldown[SPELLS.HOLY_WORD_SANCTIFY.id] = 60 - ((this.combatants.selected.traitsBySpellId[SPELLS.HALLOWED_GROUND_TRAIT.id] || 0) * SPELLS.HALLOWED_GROUND_TRAIT.coeff);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (HOLY_WORD_SPELL_IDS.indexOf(spellId) !== -1) {
+      this.holyWordCooldowns[spellId] = event.timestamp + this.holyWordMaxCooldown[spellId] * 1000;
+      this.casts[spellId] += 1;
+
+      // We are giving a buffer of 75% of CD due to the fact that the large
+      // majority of players would intentionally use spells to push holy words
+      // off cooldown if needed. This is a feelycraft metric that is open for
+      // modification or outright removal depending on opinions.
+      this.fullOvercast[spellId] += Math.floor(this._tempOvercast[spellId] / 1000 / (this.holyWordMaxCooldown[spellId] * 0.75));
```

Comment:
All Suggestions are basically feelycraft metrics, but this one may be misguided. As a player reading the suggestion that I missed X casts, it wouldn't be obvious to me that you actually mean I missed 0.75 * X casts. Why aren't you just using 1.0 here? Isn't that what a 'full' overcast is? Maybe I'm missing an element of Holy Priest play, but it seems like this is just a 'cast efficiency' suggestion that you got fancy with.

Reply:
It's similar to how I'm filtering out >80% OH Sanctify hits as being "ineffective". Because of how Serendipity works, it is possible to push spells out faster than its original CD. 

For example, You can sanctify, but realize you need it 38 seconds later. A good player would cast it and try to get it off CD for 38 seconds later while a bad one might hold the spell for 38 seconds despite being able to get it off CD for both events.

Reply:
Hmm, so you're trying to capture part of what's missing in cast efficiency for healing CDs, namely that holding for a several seconds each time when it comes off CD is fine, but holding for a few full CDs is not fine.

Maybe the spin the 0.75 into a constant at the top of the module, and give some indication in the suggestion text of how you arrived at the conclusion they were holding Holy Words too long.